sklearn/cluster/k_medoids_.py

```diff
@@ -0,0 +1,333 @@
+# -*- coding: utf-8 -*-
+"""K-medoids clustering"""
+
+# Authors: Timo Erkkilä <timo.erkkila@gmail.com>
+#          Antti Lehmussola <antti.lehmussola@gmail.com>
+#          Kornel Kiełczewski <kornel.k@plusnet.pl>
+# License: BSD 3 clause
+
+import warnings
+
+import numpy as np
+
+from ..base import BaseEstimator, ClusterMixin, TransformerMixin
+from ..metrics.pairwise import pairwise_distances
+from ..utils import check_array, check_random_state
+from ..utils.validation import check_is_fitted
+
+
+class KMedoids(BaseEstimator, ClusterMixin, TransformerMixin):
+    """
+    k-medoids class.
+
+    Read more in the :ref:`User Guide <k_medoids>`.
+
+    Parameters
+    ----------
+    n_clusters : int, optional, default: 8
+        How many medoids. Must be positive.
+
+    distance_metric : string, optional, default: 'euclidean'
+        What distance metric to use.
+
+    init : {'random', 'heuristic'}, optional, default: 'heuristic'
+        Specify medoid initialization method. Random randomly selects
+        n_clusters elements from the dataset, while heuristic picks n_clusters
```

Comment:
"n_clusters first ..." -> "the n_clusters points with the smallest sum distance to every other point". Although I'm not yet convinced this is a good heuristic.